### PR TITLE
Feat: V1.701 Adding support for computed-model loading

### PR DESCRIPTION
**Pull request by piehld**
- Updates to handle comp model local and global scores (updates by BV @brindakv)
- Updates to make use of internal computed-model identifiers
- Add testRepoHoldingsRemoteDataPrep test case

### CODE REVIEW

**Comment by piehld** [comment]
@brindakv, I just created a new branch from master and then copied your changes into my branch manually. Trying to resolve the issues with git would have led to even more complication, so this was the easiest approach. After finalizing this merge, you'll need to create a _new_ personal dev branch from the `staging-to-master` branch and work from that. If you try to use your previous dev branch, the commit history will cause all the same conflicts we just ran into.

File: rcsb/db/mongo/PdbxLoader.py [hunk 1]
```diff
@@ -783,8 +787,18 @@ def loadWorker(self, dataList, procName, optionsD, workingDir):
...
+            #
             for container in containerList:
+                # -- Assign temporary attribute with internal identifier for computed models
+                if databaseName in ["pdbx_comp_model_core"]:
+                    eObj = container.getObj("entry")
+                    sourceId = eObj.getValue("id", 0)
+                    compModelInternalId = compModelIdMapD.get(sourceId, sourceId)
+                    if sourceId == compModelInternalId:
+                        logger.warning("Unable to map computed-model sourceId (%s) to internal identifier. Will use original source ID as internal ID (rcsb_id).", sourceId)
+                    eObj.appendAttribute("rcsb_comp_model_id")
```
Comment (brindakv) [inline rcsb/db/mongo/PdbxLoader.py:799]: Perhaps better to check if the attribute exists before appending.

File: rcsb/db/mongo/PdbxLoader.py [hunk 2]
```diff
@@ -783,8 +787,18 @@ def loadWorker(self, dataList, procName, optionsD, workingDir):
...
                     cIdD[cId] = locatorObj
                     containerList.extend(cL)
-            # -- Apply methods to each container -
+            #
             for container in containerList:
+                # -- Assign temporary attribute with internal identifier for computed models
+                if databaseName in ["pdbx_comp_model_core"]:
+                    eObj = container.getObj("entry")
+                    sourceId = eObj.getValue("id", 0)
+                    compModelInternalId = compModelIdMapD.get(sourceId, sourceId)
```
Comment (brindakv) [inline rcsb/db/mongo/PdbxLoader.py:796]: Is the conversion to internalId handled by the RepositoryProvider?

**Review by brindakv — COMMENTED** [review]
@piehld Looks good me. Please see my comments.

File: rcsb/db/mongo/PdbxLoader.py (hunk 2 above)
Comment (piehld) [reply rcsb/db/mongo/PdbxLoader.py:796]: Yes, I created a new method here: https://github.com/rcsb/py-rcsb_utils_repository/blob/992480bfdefa0ad18f57bbb947a6038dc7468de2/rcsb/utils/repository/RepositoryProvider.py#L1243
I had asked Li to review that PR earlier, and just a moment ago I realized I had to make a very small bug fix so just pushed that too.

File: rcsb/db/mongo/PdbxLoader.py (hunk 1 above)
Comment (piehld) [reply rcsb/db/mongo/PdbxLoader.py:799]: OK, I can add a check and if it is already present, log an error. In normal flow, that attribute shouldn't already exist at this stage (the initial file reading stage), but if it does for whatever reason I agree it'd be good to catch that.

**Pull request merged**